ir/ir.rs

```diff
@@ -348,8 +346,24 @@ impl<'a> Iterator for Iter<'a> {
 
 /// An item's unique identifier.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct Id(u32);
-
+pub struct Id(u32, u32);
+impl Id {
+    /// create `Id` for a the given section
+    pub fn section(section: usize) -> Id {
+        assert!(section < u32::MAX as usize);
+        Id(section as u32, u32::MAX)
+    }
+    /// create `Id` for a given entry if a given section
+    pub fn entry(section: usize, index: usize) -> Id {
+        assert!(section < u32::MAX as usize);
+        assert!(index < u32::MAX as usize);
+        Id(section as u32, index as u32)
+    }
+    /// the root index
```

Comment:
Nitpick: please capitalize and punctuate doc comments' sentences, here and elsewhere. Also, newlines between methods.

Also, should note that this is the *meta* root's id, since there can be more than one non-meta root.